docs/style-guide/enso.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: style-guide
+title: Enso Style Guide
+category: style-guide
+tags: [style-guide]
+order: 7
+---
+
+# Enso Style Guide
+
+This style guide is aimed at developers writing Enso Libraries Code in the Enso
+language. It is not aimed at users writing Enso code via the graphical editor.
+
+# Encapsulation
+
+Encapsulation allows developers to present a consistent interface that is
+independent of its internal implementation. It is the foundation of reuseable
+"pieces" of code and funcationality. It allows other users (both internal and
+external) effectivley re-use components of your code.
+
+# Types of Code Components in the Enso Langauge
+
+In order to be able to talk about how we encapsulate components of code we first
+need to define what those components are. Enso has four levels of code
+components that can be built and shared with other developers. They are:
+
+- Libaries
+- Modules
+- Types
+- Methods
+
+TODO - defintion of each of these.
+
+# Public verus Private
+
+Now we have defined our components we need to look at how we define our public
+APIs for each of these and how we hide our private immplementations.
+
+## Public
+
+By default the Enso langauge is a very public one. If you don't use any of the
+private keywords and/or conventions described belown then everything is public.
+This is bad for encapsulation as other users and developers trying to re-use the
+code you wrote won't have a clean view of your code's API.
+
+## Types of Private
+
+There are a number of different ways to mark things Private in Enso
+
+### Hidden from the Graphical Editor Private
```

Comment:
- although important from _visual perspective_
- such a comment means nothing to the Enso language
- it is ignored by the runtime
- it is _just a hint_ to the IDE to not provide _visual suggestions_ to use such an API element
- as such: always prefer real `private` honored by the Enso language!